src/AccessibilityTranslator.ts

```diff
@@ -92,13 +78,28 @@ export default class AccessibilityTranslator extends React.Component<Props> {
    * drawing a frame) to make sure that the canvas and accessibility DOM are in sync.
    */
   update(callback = _.noop): void {
-    this.forceUpdate(callback);
```

Comment:
I think this is where we want to start the tree traversal and use breadth first, rather than depth first, traversal. We'd just need to mark an AccessibilityObject's parent for update (and so on up the tree) to know which branches and leaves (nodes) need updating. It _should_ simplify this logic quite a bit...